rfc/kubernetes-ui-enhancements/kubernetes-ui-enhancements.md

```diff
@@ -0,0 +1,262 @@
+
+This can be addressed by:
+- Re-using the existing UI panel to surface metadata on utilization, replica counts, and scaling events.
+- Visually indicating in the Clusters tab when a Deployment, Replica Set, or Stateful Set is targeted by a HPA. User interaction with the visual indicator will display the UI panel for HPA metadata. The visual indicator will be represented with an icon from the existing Font Awesome set available in deck. Options include:
+- [Autoprefixer](https://fontawesome.com/icons/autoprefixer?style=brands)
+- [Balance Scale](https://fontawesome.com/icons/balance-scale-left?style=solid)
+
+![](hpa-mock.png)
+*A mockup of the Spinnaker UI with an icon indicating that the Kubernetes object is associated with a Horizontal Pod Autoscaler*
+
+#### Vertical Pod Autoscalers
+
+Vertical Pod Autoscalers can be used to provide suggested values for a pod’s CPU and memory requests. VPAs can also automatically apply these recommended values. Similar to HPAs, VPAs can target workloads. there are enough conceptual similarities to extend the aforementioned HPA UI patterns to VPAs and display them in Spinnaker.
+
+### Persistent Volumes
+
+#### Current Functionality
+
+Although Spinnaker supports the deployment of Persistent Volume Claims, both Persistent Volumes and their associated claims are not rendered in the UI.
+
+#### Non-spinnaker Tooling
+
+Other Kubernetes UIs render Persistent Volume Claims and display the association to the underlying volume. This allows users to easily discern the status, access mode, and storage class of a given PVC.
+
+The [Kubernetes Dashboard tool](https://github.com/kubernetes/dashboard) also offers an extensive breakdown of the properties of the underlying volume. A PVC’s backing volume can be examined to determine its reclamation policy, source disk, and storage capacity.
+
+Finally, pods with associated Persistent Volume Claims display a link to the details of the PVC.
+
+#### Proposal
+
+The current UI offers no guidance on the relationship between a Pod, Persistent Volume Claim, and backing volume. The absence of PVCs, underlying volumes, and PVC pod links presents the opportunity to improve users understanding and enhance their ability to debug storage objects from within Spinnaker.
+
+The following set of changes are advised:
+
+- Render Persistent Volume Claims in the infrastructure view of Spinnaker’s UI
+  - Expose PVC specific metadata covering the access mode, storage class, and a reference to the underlying volume.
+  - Introduce a “Storage” infrastructure tab for rendering cluster level objects such as PVCs.
+  - Link to the underlying Persistent Volume for each PVC.
+- Render Persistent Volumes under the new Storage tab
+  - Expose Persistent Volume specific metadata on the reclamation policy, source disk, and storage capacity.
+  - Link back to the underling PVC.
+- Update the Pod UI to link to the associated PVC
+
+![](storage-mock.png)
```

Comment:
Just wanted to call out that there is an alternate scheme used to add additional tabs when "Canary" is enabled for a given application:

Before:
![image](https://user-images.githubusercontent.com/8437718/73002744-801c8280-3dd2-11ea-94b6-856b32bda21d.png)


After:
![image](https://user-images.githubusercontent.com/8437718/73002676-667b3b00-3dd2-11ea-8ac6-c92621891114.png)
![image](https://user-images.githubusercontent.com/8437718/73002825-9d515100-3dd2-11ea-8f4a-baa9a56f853b.png)

Maybe we can employ some alternate grouping for these newly-rendered resources as well.

Reply:
Thanks, I hadn't seen the Canary specific UI. I'm going to take a look at it.

Reply:
@duftler and I chatted about this. The Canary UI relabels the Pipelines link and turns it into a menu. In this case, the infrastructure view will continue functioning as a menu and additional menu items will be added for Storage and Config.